go/kernel/posix/socket.go

```diff
@@ -83,6 +83,12 @@ func (k *PosixKernel) Setsockopt(fd co.Fd, level, opt int, valueIn co.Buf, size
 	return 0
 }
 
+func (k *PosixKernel) Getsockname(fd co.Fd, sa syscall.Sockaddr, size co.Len) uint64 {
+	opt, err := syscall.Getsockname(int(fd))
+	sa = opt
```

Comment:
I'm confused, does `sa = opt` actually return the socket to the client? It's too implicit, so if it does work we should fix that. I'd expect you to replace `sa` with a `kernel/common.Obuf` (output buffer, and I usually import common as `co`), and do `obuf.Pack(sa)`, which would look like this:

```
Getsockname(fd co.Fd, co.Obuf, size co.Len) {
    sa, err := syscall.Getsockname(int(fd))
    if err != nil {
        return Errno(err)
    }
    if err := obuf.Pack(sa); err != nil {
        return UINT64_MAX // FIXME
    }
    return 0
}
```

(The `UINT64_MAX` is because I don't currently have any way in Usercorn to retrieve real kernel enums, which can vary per kernel, so I can't return real error numbers yet... in the long run, I want to add a second optional `error` or `errno` return from syscalls and interpret that in the syscall dispatcher / arch module)